Fix the gray-out bug in the nouveau DRI2 driver by disabling nouveau …

martinwguy opened the pull request:
…DRI2

With certain nVidia cards, https://github.com/login flashes the login box
twice and then makes the tab all gray and gmail.com's login box flashes
between the right thing and all gray repeatedly.
This seems to be a bug in the nouveau DRI2 driver; this disables that so
X starts up with the nouveau driver without DRI2 accelleration.
See https://github.com/sugarlabs/sugar-live-build/issues/20

martinwguy committed:
Fix the gray-out bug in the nouveau DRI2 driver by disabling nouveau DRI2

With certain nVidia cards, https://github.com/login flashes the login box
twice and then makes the tab all gray and gmail.com's login box flashes
between the right thing and all gray repeatedly.
This seems to be a bug in the nouveau DRI2 driver; this disables that so
X starts up with the nouveau driver without DRI2 accelleration.
See https://github.com/sugarlabs/sugar-live-build/issues/20

quozl commented:
As this affects all use of Debian Live Build, it would be best to get it fixed in Debian, that way when the fix is made we'll get the fix here automatically.  What's the status of the issue in Debian?  (i.e. fixed in stable with an update, fixed in testing, fixed in experimental, or not fixed).

martinwguy commented:
On 15/12/2020, James Cameron <notifications@github.com> wrote:
> As this affects all use of Debian Live Build, it would be best to get it
> fixed in Debian, that way when the fix is made we'll get the fix here
> automatically.  What's the status of the issue in Debian?  (i.e. fixed in
> stable with an update, fixed in testing, fixed in experimental, or not
> fixed).

I sent a message to the maintainers' list about it, but got no replies.
https://alioth-lists.debian.net/pipermail/pkg-gnome-maintainers/2020-November/162142.html
That may be because the list is high-volume with mostly
automatically-posted messages.
If someone would like the fag of posting a proper Debian bug, against
metacity, that might help better. It may also be worth prodding the
Debian Sugar maintainers, as it affects their product and that might
provide some in-house leverage to get the fix it into buster-updates.

    M

quozl commented:
Thanks, but this issue is not about metacity.  I'll take your reply as relating to issue https://github.com/sugarlabs/sugar-live-build/pull/30#issuecomment-746319398 though, and that's useful knowledge.

Please reply again for this issue; what are your thoughts about this issue in Debian?

martinwguy commented:
You're right, sorry. Confusion.
As you said, the real answer is to find the bug in the nouveau DRI
driver, or in the way that the graphic toolkit uses it, and fix that.
I'm not sure I'm up to that though, as the test cycle is too long.

   M

On 16/12/2020, James Cameron <notifications@github.com> wrote:
> Thanks, but this issue is not about metacity.  I'll take your reply as
> relating to issue
> https://github.com/sugarlabs/sugar-live-build/pull/30#issuecomment-746319398
> though, and that's useful knowledge.
>
> Please reply again for this issue; what are your thoughts about this issue
> in Debian?
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/sugarlabs/sugar-live-build/pull/29#issuecomment-746984627

quozl commented:
Thanks.  I'll take that as the status of the issue in Debian is not known.

martinwguy commented:
...unless someone wants to file an urgent Debian bug report. Me, I'm
tired of doing such, but if someone wants to take the hassle...

   M

quozl commented:
Thanks.  I can't file a report because I can't reproduce it; I don't have the hardware.  It has to be someone with the hardware, but you are unavailable.

We don't know the status of the issue in Debian; we know it doesn't work for you on i386 on Debian Stable, but we don't know what happens with Debian Testing or Debian Experimental.  There's a possibility it is fixed already.

Consider the possibility.  `/usr/lib/i386-linux-gnu/dri/nouveau_dri.so` is provided by the `liggl1-mesa-dri` package, which are the OpenGL API DRI modules.  The source package is `mesa`.  It is version 18.3.6 on Debian Stable.  It is version 20.2.4 on Debian Testing.  It is version 20.3.0 on Debian Experimental.  There are no Nouveau specific entries in the [Debian changelog](https://salsa.debian.org/xorg-team/lib/mesa/-/blob/debian-unstable/debian/changelog).  The source package has about [14 bugs that mention Nouveau](https://bugs.debian.org/cgi-bin/pkgreport.cgi?src=mesa), though none of them seem to exactly match what you describe.  Comparing 18.3.6 with 20.3.0 for the `src/mesa/drivers/dri/nouveau` shows some functional changes that may be significant.  Reviewing commit messages of all the patches that touch this driver between those two versions, and I come away with;

* regression may have been before 18.3.6,
* almost all the changes are to common code,
* the project accepts commits without comprehensive testing, they are reliant on testing by others, so it is reasonable that for a little-used configuration (i386) they may not have noticed what was broken,
* [8c5c133741f](https://gitlab.freedesktop.org/mesa/mesa/-/commit/8c5c133741f) ("mesa: call Driver.TexParameter in glPopAttrib to fix r100, r200, old nouveau") may be a fix, which was part of https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/8046 and looks to have been released in 20.3.1

So I try to discern why I should merge this.  Can you test later Mesa?  Is this problem i386 specific?

martinwguy commented:
On 18/12/2020, James Cameron <notifications@github.com> wrote:
> I can't file a report because I can't reproduce it; I don't have
> the hardware.  It has to be someone with the hardware, but you are
> unavailable.

I don't think I am. Yes, I agree it has to be reported by someone with
the hardware, and it would be impossible for any Debianite to
investigate the bug in the Debian package unless they have access to
the affected hardware. Similarly, I doubt a Debian Developer would
accepthe bug report or a purported patch unless they could personally
verify the existence and removal of the bug, and despite having done
most of the Debian armel port, I did that without becoming a DD.

> we know it doesn't work for you on i386 on Debian Stable,
> but we don't know what happens with Debian Testing.
> Is this problem i386 specific?

I'm working on trying SLS with Debian testing/bullseye.

We don't know whether it's 32-bit-dependent or not, as I can't
transplant that video chip into a 64-bit computer. We also don't know
whether it is specific to one chip family or to all nVidia systems as
I don't have any other nVidia video chips and haven't received reports
from anyone who does.

It may be that it only affects certain chips supported by the
"nouveau" accelerator, and that the affected chips are from the
pre-64-bit era, which would make it only affect 32-bit systems as a
coincidence.

If it turns out to be fixed in the next Debian release, I suggest that
the remove-nouveau-acceleration fix will do for SLB in the meantime. I
don't see much point in having video acceleration enabled for a video
chip whose VA doesn't work.

> So I try to discern why I should merge this.

To make it work for all the people who have similarly affected chips.

> Can you test later Mesa?
That's an idea. Thanks for the pointers

     M
